### PR TITLE
feat: Provide cwd for che terminal creation

### PR DESCRIPTION
**Pull request by RomanNikitenko**
### What does this PR do?
Provides current working directory for a terminal creation when user uses `New Terminal (Select a container) action`:
 - value of the `PROJECTS_ROOT` env variable if there is no any workspace folder
 - path to a folder if there is only one workspace folder
 - IDE proposes user to select a folder if there is more then 1 workspace folder

### What issues does this PR fix?

https://github.com/eclipse-che/che/issues/23022

#### Empty workspace use case
- Go to the dashboard => `Create Workspace` page
- `Use an Editor Definition` => `Editor Image` => set `quay.io/che-incubator-pull-requests/che-code:pr-384-amd64`
- Select `Empty Workspace` to start a workspace
- main menu => `Terminal` => `New Terminal (Select a container)`
- check that `/projects` is used as a current working directory: type `pwd` for example in the created terminal

#### Workspace contains only one workspace folder
- Go to the dashboard =>`Create Workspace` page
- `Use an Editor Definition` => `Editor Image` => set `quay.io/che-incubator-pull-requests/che-code:pr-384-amd64`
- Select any sample to start a workspace
- main menu => `Terminal` => `New Terminal (Select a container)`
- check that project path is used as a current working directory: type `pwd` for example in the created terminal

#### Workspace contains more then one workspace folder
- Go to the dashboard =>`Create Workspace` page
- `Git repo URL` => set `https://github.com/RomanNikitenko/web-nodejs-sample/tree/test-two-projects`
- main menu => `Terminal` => `New Terminal (Select a container)`
- Select any folder when IDE proposes it
- check that selected path is used as a current working directory: type `pwd` for example in the created terminal

### Does this PR contain changes that override default upstream Code-OSS behavior?
- [ ] the PR contains changes in the [code](https://github.com/che-incubator/che-code/tree/main/code) folder (you can skip it if your changes are placed in a che extension )
- [ ] the corresponding items were added to the [CHANGELOG.md](https://github.com/che-incubator/che-code/blob/main/.rebase/CHANGELOG.md) file
- [ ] rules for automatic `git rebase` were added to the [.rebase](https://github.com/che-incubator/che-code/tree/main/.rebase) folder

==========================================
Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

File: code/extensions/che-terminal/src/extension.ts [hunk 1]
```diff
@@ -127,3 +128,29 @@ export class MachineExecPTY implements vscode.Pseudoterminal {
 
 export function deactivate(): void {
 }
+
+/**
+ * Provides current working directory for a terminal creation:
```
Comment (vitaliy-guliy) [inline code/extensions/che-terminal/src/extension.ts:133]:
```suggestion
 * Provides current working directory for a terminal creation.
```

File: code/extensions/che-terminal/src/extension.ts [hunk 2]
```diff
@@ -127,3 +128,29 @@ export class MachineExecPTY implements vscode.Pseudoterminal {
...
+	if (folders === undefined || folders.length < 1) {
+		return process.env.PROJECTS_ROOT as string;
+	}
+
+	if (folders.length === 1) {
+		return folders[0].uri.path
+	}
+
+	const options = {
+		placeHolder: "Select current working directory for new terminal"
```
Comment (vitaliy-guliy) [inline code/extensions/che-terminal/src/extension.ts:152]: I would change the message to
```suggestion
		placeHolder: "Select a working directory for the new terminal"
```

**Review by vitaliy-guliy — APPROVED** [review]
I have just tested the behavior for the described cases and can confirm the improvement works as expected.

Only one thing I would fix in the next PR is that if you clicked the `New Terminal (Select a Container)` menu item you are proposed to select a working directory. It is confusing a bit.

So if we have only one container, I would change the menu item title to `New Terminal in a Container` or `New Terminal in a Tooling Container` or something like that.

File: code/extensions/che-terminal/src/extension.ts (hunk 2 above)
Comment (RomanNikitenko) [reply code/extensions/che-terminal/src/extension.ts:152]: I prefer to leave it as is.
VS Code uses the same message

Comment (RomanNikitenko) [reply code/extensions/che-terminal/src/extension.ts:152]: Please see, for example, here  https://github.com/microsoft/vscode/blob/38623f21e7de9d8801a2c1eb7eeda514398a3e77/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L1810

**Comment by RomanNikitenko** [comment]
@vitaliy-guliy 
thank you for review and testing my changes!

about 
> if we have only one container, I would change the menu item title to New Terminal in a Container or New Terminal in a Tooling Container or something like that.

Thanks for your idea for the improvement!
I checked what we can do with it and see that:
- currently the title is placed on the VS Code browser side
- info about available containers is available on the che-extension side
- we should think about localization as well - as we have request for adding it from users

So - currently I don't see some easy way to implement it, probably we could find more suitable name for the command...

**Comment by RomanNikitenko** [comment]
@azatsarynnyy @vitaliy-guliy 
- I fetched all changes from the main branch to my branch
- tested different use cases
- I didn't find any regression - everything works as described in the PR description

So, I'm going to merge the current PR.

**Pull request merged**